### PR TITLE
Adds support for Anthropic's messages API

### PR DESCRIPTION
**Pull request by svs**
I saw @obie's PR and improved on it slightly.

- I removed the use of `present?` which is a Rails-only method so the library works outside of Rails as well
- I removed the retry logic which I believe should be a concern of the caller. This gem should faithfully report whatever Anthropic API returns
- I am also not a fan of having the `messages` function API break out the parameters. I understand it makes it marginally easier to call the API but there should also be a 1:1 mapping between the Anthropic API docs and our gem, which @obie's API breaks. 
- I added a spec for the new method.

Hope this helps.

### CODE REVIEW

File: spec/anthropic/client/messages_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+RSpec.describe Anthropic::Client do
+  describe "#complete" do
```
Comment (brucruz) [inline spec/anthropic/client/messages_spec.rb:2]: typo here: should be "#messages"

Comment (svs) [reply spec/anthropic/client/messages_spec.rb]: fixed.

**Review by brucruz — APPROVED** [review]

**Comment by deepakmahakale** [comment]
This is nice 👍

File: spec/anthropic/client/messages_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+RSpec.describe Anthropic::Client do
+  describe "#messages" do
+    context "with a prompt and max_tokens", :vcr do
+      let(:prompt) { "How high is the sky?" }
+      let(:max_tokens) { 5 }
+
+      let(:response) do
+        Anthropic::Client.new(access_token: ENV['ANTHROPIC_API_KEY']).messages(
+          parameters: {
+            model: model,
```
Comment (deepakmahakale) [inline spec/anthropic/client/messages_spec.rb:10]: @svs Do we need to pass this from somewhere?

Comment (svs) [reply spec/anthropic/client/messages_spec.rb:10]: Please see https://github.com/alexrudall/anthropic?tab=readme-ov-file#release

Comment (deepakmahakale) [reply spec/anthropic/client/messages_spec.rb:10]: I meant I don't see a reference for `model` 
Do we need to pass or assign `model` with `"claude-2"`

**Comment by duncantmiller** [comment]
@alexrudall Looks great to me thanks for the work all!

**Comment by rameerez** [comment]
I've also tested this, works fine!

Thanks for the work, this makes it possible to use Claude 3, since the `complete` API was deprecated in favor of `messages` for the new models.

For future reference / Google, the error Anthropic throws when you try to use Claude 3 on the `complete` API is this:
```
{"type"=>"error", "error"=>{"type"=>"invalid_request_error", "message"=>"\"claude-3-opus-20240229\" is not supported on this API. Please use the Messages API instead."}}
```

**Comment by frmsaul** [comment]
Looks great! Thanks so much

**Comment by frmsaul** [comment]
We should probably update the README as well.

**Comment by deepakmahakale** [comment]
@alexrudall I have added documentation for messages API in this PR - https://github.com/alexrudall/anthropic/pull/12

We can merge #12 once this pr is merged

**Comment by Maclenn77** [comment]
I'm using this branch for a project, it works well.

**Comment by drnic** [comment]
> We should probably update the README as well.

Yep, let's get claude-3 + `messages` to top of README.

**Comment by drnic** [comment]
@alexrudall perhaps make some other people maintainers/gem releases?

**Comment by andreibondarev** [comment]
Let's ship this! 🚢 😄

**Comment by bloolizard** [comment]
nice can you guys merge this in!

**Comment by alexrudall** [comment]
Released in [v0.2](https://github.com/alexrudall/anthropic/releases/tag/v0.2.0) - thanks all for your contributions, esp @obie and @svs !

**Pull request merged**